lib/matplotlib/path.py

```diff
@@ -1042,3 +1203,248 @@ def get_path_collection_extents(
+    # $\phi$ is the incidence angle of the corner's bisector. Notice that in
+    # the limit ($\phi = \theta/2$) where the "corner" is flush with the bbox,
+    # this correctly simplifies to just $w/2$.
+    elif joinstyle == 'miter':
+        # matplotlib currently doesn't set the miterlimit...
+        _nominal_miter_limit = 10  # pdf and agg do this
+        if 1/np.sin(theta/2) > _nominal_miter_limit:
+            return _stroke_x_overflow(width, phi, theta, 'bevel', capstyle)
+        else:
+            return (width/2)*np.sin(phi)/np.sin(theta/2)
+    # a beveled edge is exactly the convex hull of its two composite lines with
+    # capstyle='butt'. So we just compute the individual lines' incidence
+    # angles and take the maximum of the two padding values
+    elif joinstyle == 'bevel':
+        phi1 = phi + theta/2
+        phi2 = phi - theta/2
+        return (width/2) * max(np.abs(np.cos(phi1)), np.abs(np.cos(phi2)))
+    # finally, _joinstyle = "round" is just _joinstyle = "bevel" but with
+    # a hemispherical cap. we could calculate this but for now no markers use
+    # it....except those with "no corner", in which case we can treat them the
+    # same as squares...
+    elif joinstyle == 'round':
+        return width/2  # hemispherical cap, so always same padding
+    else:
+        raise ValueError(f"Unknown joinstyle: {joinstyle}")
+
+
+def _pad_extents_stroked_vertex(extents, vinfo, markeredgewidth, joinstyle,
+                                capstyle):
+    """
+    Accumulator for building true extents from `.VertexInfo`s.
+
+    Parameters
+    ----------
+    extents : (4,) array_like of float
+        The extents (xmin, ymin, xmax, ymax) of the `~.transforms.Bbox` of the
+        vertices. Modified in place so that the corner described by *vinfo*
+        fits into the extents when stroked with a width of *markeredgewidth*.
+    vinfo : `.VertexInfo`
+        Information about the corner or cap at one vertex.
+    markeredgewidth : `float`
+        The width of the stroke being drawn.
+    joinstyle : {'miter', 'bevel', 'round'}
+        How line corners are to be drawn.
+    capstyle : {'butt', 'round', 'projecting'}
+        How line ends are to be drawn.
+
+    Notes
+    -----
+    Implemented by wrapping `._stroke_x_overflow`. This function checks which
+    direction the corner (or cap) is pointing, then for each side of *extents*
+    that might need padding, it rotates the corner to point in the positive x
+    direction and calls `._stroke_x_overflow` to get the padding.
+    """
+    xmin = 0
+    ymin = 1
+    xmax = 2
+    ymax = 3
+    # now for each direction (up/down/left/right), convert the absolute
+    # incidence angle into the incidence angle relative to that respective side
+    # of the bbox, and see if the stroked vertex expands the extents...
+    x, y = vinfo.apex
+    if np.cos(vinfo.incidence_angle) > 0:
+        incidence_angle = vinfo.incidence_angle + np.pi/2
```

Comment:
In these cases, I'd name the local variable `theta`, as the reference for the angle is different (I think?), so it's different than `incidence_angle`.